src/pages/workshop-description/index.js

```diff
@@ -0,0 +1,232 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import Details from './styled'
+import { useState, useEffect } from 'react'
+import { cms } from '../../client'
+import {
+  faUser,
+  faPenToSquare,
+  faHandHoldingDollar,
+  faCalendarDays,
+  faLocationDot,
+  faRectangleXmark,
+  faBullhorn
+} from '@fortawesome/free-solid-svg-icons'
+import { Swiper, SwiperSlide } from 'swiper/react'
+import { FreeMode, Navigation, Thumbs } from 'swiper/modules'
+import env from 'react-dotenv'
+import 'swiper/css'
+import 'swiper/css/free-mode'
+import 'swiper/css/navigation'
+import 'swiper/css/thumbs'
+import { useParams } from 'react-router'
+import { Link } from 'react-router-dom'
+
+export const WorkshopDetails = () => {
+  const [thumbsSwiper, setThumbsSwiper] = useState(null)
+  const [workshops, setWorkshops] = useState([])
+  const [galleryPhoto, setGalleryPhoto] = useState([])
+  const { id } = useParams()
+  useEffect(() => {
+    cms
+      .get(`api/workshops/${id}/?populate=parceires&populate=foto_oficina`)
+      .then((response) => {
+        const { data } = response.data
+        const photoWorkshop = data.attributes.foto_oficina
+        const images = photoWorkshop.data.map((image) => {
+          return {
+            id: image.id,
+            name: image.attributes?.name,
+            url: env.URL_CMS + image.attributes?.url
+          }
+        })
+        setGalleryPhoto(images)
+        setWorkshops(data)
+      })
+  }, [])
+  const daysOfWeekInPtBr = [
+    'Domingo',
+    'Segunda',
+    'Terça',
+    'Quarta',
+    'Quinta',
+    'Sexta',
+    'Sábado'
+  ]
+  const formatDate = (date) => {
+    const day = date.toLocaleDateString(undefined, {
+      day: 'numeric',
+      Timezone: 'UTF'
+    })
+    const month = date.toLocaleDateString('pt-BR', { month: 'short' })
+    const year = date.toLocaleDateString(undefined, {
+      year: 'numeric',
+      Timezone: 'UTF'
+    })
+    return `${daysOfWeekInPtBr[date.getDay()]}, ${day} ${month} ${year}`
+  }
+  const formatWorkshopDuration = (workshop) =>
+    `, ${workshop.attributes?.horario_inicio} > ${workshop.attributes?.horario_fim}`
+
+  const endsOnSameDay = (workshop) =>
+    workshop.attributes?.data_inicio === workshop.attributes?.data_fim
+
+  const formatWorkshopDates = (workshop) => {
+    if (workshop && workshop.attributes && workshop.attributes.data_inicio && workshop.attributes.data_fim) {
+      return endsOnSameDay(workshop)
+        ? formatDate(new Date(workshop.attributes?.data_inicio))
+        : `${formatDate(new Date(workshop.attributes?.data_inicio))} 
+        até ${formatDate(new Date(workshop.attributes?.data_fim))}`
```

Comment:
Dado que isso aqui so executa se o if for verdadeiro, logo, significa que attributes existe e não é nulo, portanto, não precisamos de null-coalescing aqui:

```suggestion
        ? formatDate(new Date(workshop.attributes.data_inicio))
        : `${formatDate(new Date(workshop.attributes.data_inicio))} 
        até ${formatDate(new Date(workshop.attributes.data_fim))}`
```